src/main/java/com/tfgm/services/TramStopService.java

```diff
@@ -0,0 +1,156 @@
+      JSONObject tempStop = allTramStopData.getJSONObject(i);
+      tramStopHashMap.put(
+          tempStop.getString("tramStopName"),
+          new TramStop(
+              tempStop.getString("location"),
+              tempStop.getString("direction"),
+              tempStop.getString("line")));
+    }
+    for (int i = 0; i < allTramStopData.length(); i++) {
+      JSONObject tempStop = allTramStopData.getJSONObject(i);
+
+      TramStop currentTramStop = tramStopHashMap.get(tempStop.getString("tramStopName"));
+
+      currentTramStop.setPrevAndNextStops(
+          tempStop.getJSONArray("prevStop").toList().stream()
+              .map(tramStopHashMap::get)
+              .map(TramStopContainer::new)
+              .toArray(TramStopContainer[]::new),
+          tempStop.getJSONArray("nextStop").toList().stream()
+              .map(tramStopHashMap::get)
+              .map(TramStopContainer::new)
+              .toArray(TramStopContainer[]::new));
+    }
+
+    for (TramStop tramStop : tramStopHashMap.values()) {
+
+      System.out.println(tramStop.toString());
+      for (TramStopContainer n : tramStop.getNextStops()) {
+        TramStopContainer nextStopPrevReference =
+            Arrays.stream(n.getTramStop().getPrevStops())
+                .filter(s -> s.getTramStop().equals(tramStop))
+                .toList()
+                .get(0);
+
+        nextStopPrevReference.setTramLinkStop(n.getTramLinkStop());
+      }
+    }
+
+    // MAKE SURE YOU REMOVE APOSTROPHES AND WHITESPACE WHEN FINDING TRAMSTOP
+    for (TramStop i : tramStopHashMap.values()) {
+      System.out.println("\n\n" + i.getStopName() + i.getDirection());
+      System.out.println(i);
+
+      System.out.println("\nNextStop:");
+      for (TramStopContainer n : i.getNextStops()) {
+        System.out.println(n.getTramLinkStop());
+      }
+      System.out.println("\nPrevStop:");
+      for (TramStopContainer n : i.getPrevStops()) {
+        System.out.println(n.getTramLinkStop());
+      }
+    }
+  }
+
+  public void update() throws URISyntaxException, IOException {
+    HttpClient httpclient = HttpClients.createDefault();
+
+    URIBuilder builder = new URIBuilder("https://api.tfgm.com/odata/Metrolinks");
+    // "StationLocation":"Withington","AtcoCode":"9400ZZMAWIT2","Direction":"Outgoing","Dest0":"East
+    // Didsbury","Carriages0":"Single","Status0":"Due","Wait0":"3","Dest1":"East
+    // Didsbury","Carriages1":"Double","Status1":"Due","Wait1":"14","Dest2":"East
+    // Didsbury","Carriages2":"Single","Status2":"Due","Wait2":"16","Dest3":"","Carriages3":"","Status3":"","MessageBoard":"Welcome to Metrolink. For up to date travel information contact Metrolink twitter on @MCRMetrolink or visit www.TfGM.com. Take care of each other Manchester.","Wait3":"","LastUpdated":"2023-05-26T15:29:19Z"
+
+    URI uri = builder.build();
+    HttpGet request = new HttpGet(uri);
+    request.setHeader("Ocp-Apim-Subscription-Key", "810dfa20bbd44815b07d8cb6f9d3be96");
```

Comment:
So this acts as your data layer, as it's where you're requesting data from.Add this logic into the persistence layer and add your data manipulation logic in there, then return the data in the correct format back to your service layer where you can perform business logic. 

Client request -> Controller -> Service -> Persistence (Manipulate data to be ready for service layer) ->  Service (Perform business logic) -> Controller -> Client response